wazuh/wazuh-qa
# E2E test: Monitoring Docker

|Related issues|
|---|
| #2873, #2935, #3003|

## Description

This module tests the integration of Wazuh with Docker using 1 instance of the manager that will be used by Ansible to run tasks on it.

### Design

1. Get the test cases defined in:
```
end_to_end/
└ ANY_TEST/
    └ data/
        └ playbooks/
        └ test_cases/ <--- here
	     ├ cases_file.yml
             .
             .
```

2. Configure the environment (install docker, configure wodle, etc.) using the test cases data by running some playbooks that will be stored in:
```
end_to_end/
└ ANY_TEST/
    └ data/
        └ playbooks/ <--- here
	     ├ 1.yml
             ├ 2.yml
             .
             .
```

3. Get the alerts (if they were generated) from the wazuh-indexer API
4. Check if the expected alert matched one of the alerts obtained from the API
   4.1. If not, then check if the alert at least is found in the log file
        4.1.1. If not, raise a TimeoutError
        4.1.2. If yes, raise an AssertionError

### Test cases

1. Run the `docker pull nginx` command
2. Run the `docker run -d -P --name nginx_container nginx` command
3. Run the `docker exec -ti nginx_container /bin/bash` command
4. Run the <code>docker stop `docker ps -a -q` && docker rm `docker ps -a -q`</code> command

### Run test

Precondition: Create an inventory as follows
```
all:
  hosts:
    wazuh-manager:
      ansible_connection: ssh
      ansible_user: vagrant
      ansible_ssh_private_key_file: /home/mauro/insecure_private_key
      ansible_python_interpreter: /usr/bin/python3.8
      dashboard_user: USER
      dashboard_password: PASS
```

Execute the test: <code>python -m pytest tests/end_to_end/TEST-PATH --inventory_path PATH-TO-INVENTORY</code>

### Test execution

|Local|OS|Commit|By
|--|--|--|--
| [🟢](https://github.com/wazuh/wazuh-qa/files/9004289/r1-CentOS-manager-2935.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9004290/r2-CentOS-manager-2935.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9004291/r3-CentOS-manager-2935.zip) | CentOS | https://github.com/wazuh/wazuh-qa/pull/2948/commits/010c9f5af10700f0f4060b31a88bcd43b03ccc02 | @mauromalara
